app/models/product.rb

```diff
@@ -1,7 +1,15 @@
 class Product < ApplicationRecord
   belongs_to :category
+  has_many :images, dependent: :destroy
   has_many :ratings, dependent: :destroy
   has_many :order_details, dependent: :destroy
 
   validates :name, :price, :quantity_in_stock, presence: true
+
+  scope :by_name, (lambda do |name|
+                     where("name LIKE (?)", "%#{name}%") if name.present?
+                   end)
+  scope :price_order, ->(order){order(price: order)}
```

Comment:
fix tương tự bên dưới em nhé
đặt tên param tránh trùng với các keyword dễ gây hiểu nhầm
```suggestion
  scope :order_by_price, ->(criteria){order price: criteria}
```